### PR TITLE
Use Dependabot to auto-update GitHub Actions and Python dependency versions

### PR DESCRIPTION
**Pull request by jwodder**

### CODE REVIEW

**Comment by jwodder** [comment]
@AlmightyYakob Ping.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+version: 2
+updates:
+  - package-ecosystem: pip
```
Comment (AlmightyYakob) [inline .github/dependabot.yml:3]: Since this project uses `poetry`, should this be changed?
```suggestion
  - package-ecosystem: poetry
```

Comment (jwodder) [reply .github/dependabot.yml:3]: No, Dependabot for Poetry uses the `pip` setting: https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#package-ecosystem

Comment (AlmightyYakob) [reply .github/dependabot.yml:3]: oh, my bad, I misread that page then.

**Review by AlmightyYakob — APPROVED** [review]

**Pull request merged**